### DISCUSSION TITLE
Switched to php 8.1 - the speed of work has increased a lot!

### DISCUSSION THREAD

**Discussion by MikleBas** [magicbug/Cloudlog#1815] [General]
Actually, that's all I wanted to tell you :)
If it is possible to switch to 8.1, then go, the speed of the program will increase (in my opinion) by 20-25 percent
From the heart
73!

**Comment by AndreasK79** [comment]
No errors at all? I did some testing with 8.1 before Cloudlog v2 was released, and I did get several errors. I found that Codeigniter 3 isn't fully updated to fix all PHP 8.1 changes.

**Reply by MikleBas** [reply]
Certainly not without a fly in the ointment :)
So far the only fatal error I've found is when it comes to log export (just export)
the server gives an error 500, and since I constantly use export several times a day, I temporarily rolled back to the previous version (and the previous version is no longer comfortable in terms of speed, but it is reliable, I hope that you will fix it soon)

**Reply by AndreasK79** [reply]
8.0 should work fine at least. Fixing this, is not up to me. It is up to the people working on Codeigniter, as I have too little knowledge about Codeigniter and how things should be fixed for PHP 8.1.

**Reply by MikleBas** [reply]
checked for 8.0, the same, error
When exporting, it works as quickly, only export does not work, error 500
Version 7.4 is stable

**Reply by AndreasK79** [reply]
Which export are you using when you get the error?

**Reply by MikleBas** [reply]
EQSL.CC - OK
Export as Adif Format - Error 500
The rest of the export options are not relevant to me in connection with the paid official services of their services

**Reply by AndreasK79** [reply]
Ok. I don't use eqsl, so never tested. I'll take a look tonsee if I see anything obvious.

**Reply by R1BLH** [reply]
@MikleBas switched to php 8.1. All looks fine. There in no error 500 on the export in ADIF format.

**Reply by MikleBas** [reply]
I'll try to turn on the developer mode and see the logs why the error is obtained. I'll post the results here

**Reply by MikleBas** [reply]
```
A PHP Error was encountered
Severity: 8192

Message: Return type of CI_Session_files_driver::open($save_path, $name) should either be compatible with SessionHandlerInterface::open(string $path, string $name): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

Filename: drivers/Session_files_driver.php

Line Number: 132

A PHP Error was encountered
Severity: 8192

Message: Return type of CI_Session_files_driver::close() should either be compatible with SessionHandlerInterface::close(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

Filename: drivers/Session_files_driver.php

Line Number: 298

A PHP Error was encountered
Severity: 8192

Message: Return type of CI_Session_files_driver::read($session_id) should either be compatible with SessionHandlerInterface::read(string $id): string|false, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

Filename: drivers/Session_files_driver.php

Line Number: 168

A PHP Error was encountered
Severity: 8192

Message: Return type of CI_Session_files_driver::write($session_id, $session_data) should either be compatible with SessionHandlerInterface::write(string $id, string $data): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

Filename: drivers/Session_files_driver.php

Line Number: 241

A PHP Error was encountered
Severity: 8192

Message: Return type of CI_Session_files_driver::destroy($session_id) should either be compatible with SessionHandlerInterface::destroy(string $id): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

Filename: drivers/Session_files_driver.php

Line Number: 321

A PHP Error was encountered
Severity: 8192

Message: Return type of CI_Session_files_driver::gc($maxlifetime) should either be compatible with SessionHandlerInterface::gc(int $max_lifetime): int|false, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

Filename: drivers/Session_files_driver.php

Line Number: 362

An uncaught Exception was encountered
Type: Error

Message: Call to a member function format() on bool

Filename: /www/wwwroot/MY-SITENAME/application/models/Adif_data.php

Line Number: 87
```

**Reply by MikleBas** [reply]
PHP-7.4
![php7](https://user-images.githubusercontent.com/99086477/204977828-a548439c-54e7-4748-a6b4-f380fc843f6b.png)

PHP8.1
![php8](https://user-images.githubusercontent.com/99086477/204977905-08aea25f-13f8-4068-94e1-e289288f69f4.png)

**Comment by dg9vh** [comment]
Are you on main or dev branch?

**Reply by MikleBas** [reply]
To be honest, I have no idea, I'm a simple user of the program :)
If this is important, then tell me how to check it, and I will tell you the answer to your question

**Reply by MikleBas** [reply]
On the main. It turns out that you need to turn on the Developer mode to find out more about the system and the version of the program :)

**Comment by AndreasK79** [comment]
#1843 added. If no problems pop up, PHP8.1 will be supported in next release.

**Reply by MikleBas** [reply]
Hello!
So far, there are no obvious problems. Except for the ones mentioned earlier in this post.
Thanks for the news! Look forward to! :)

**Reply by MikleBas** [reply]
PS I'll add my wish:
1. Is there some kind of checklist that would check whether all the necessary php components are on the server and whether they are suitable for the correct operation of your program?
This would remove a lot of problems with diagnosing why everything works on 7.4 and does not work on 8.1 for example.
